File: python/day2a.py
import re

sum = 0
f = open('input.txt', 'r')

while True:
    line = f.readline()
    if line =='':
        break

    a = re.split(r'[:;,]', line)
    id = 0
    check = True

    for chunk in a:
        chunk = chunk.strip()
        val = chunk.split()[0]
        col = chunk.split()[1]
        if col.isdigit():
            id = int(col)
        if val.isdigit():
            match col:
                case 'blue':
                    if int(val) > 14:
                        check = False
                case 'red':
                    if int(val) > 12:
                        check = False
                case 'green':
                    if int(val) > 13:
                        check = False
    if check:
        sum += id
print(sum)

File: c/day2a.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(void)
{
    int sum = 0;
    char line[900];
    while (fgets(line, sizeof(line), stdin))
    {
        int game, check = 1;
        sscanf(line, "Game %d:", &game);
        char delim[] = " ,;:\n";
        char *token = strtok(line, delim);
        strtok(NULL, delim);
        token = strtok(NULL, delim);
        while (token != NULL)
        {
            int val = atoi(token);
            token = strtok(NULL, delim);
            if (strcmp(token, "red") == 0 && val > 12)
                check = 0;
            else if (strcmp(token, "green") == 0 && val > 13)
                check = 0;
            else if (strcmp(token, "blue") == 0 && val > 14)
                check = 0;
            token = strtok(NULL, delim);
        }
        if (check)
            sum += game;
    }
    printf("%d\n", sum);
}